cmd/cli/namespace_test.go

```diff
@@ -93,6 +93,34 @@ var _ = Describe("Running the namespace add command", func() {
 				Expect(out.String()).To(Equal(fmt.Sprintf("Namespace [%s] successfully added to mesh [%s]\nNamespace [%s] successfully added to mesh [%s]\n", testNamespace, testMeshName, testNamespace2, testMeshName)))
 			})
 		})
+		Context("given one namespace with osm-controller installed in it as an arg", func() {
+			BeforeEach(func() {
+				out = new(bytes.Buffer)
+				fakeClientSet = fake.NewSimpleClientset()
+				// mimic osm controller deployment in testNamespace
+				deploymentSpec := createDeploymentSpec(testNamespace, defaultMeshName)
+				fakeClientSet.AppsV1().Deployments(testNamespace).Create(context.TODO(), deploymentSpec, metav1.CreateOptions{})
+
+				nsSpec := createNamespaceSpec(testNamespace, "")
+				fakeClientSet.CoreV1().Namespaces().Create(context.TODO(), nsSpec, metav1.CreateOptions{})
```

Comment:
I was surprised to see that we're creating the namespace after creating the deployment in the namespace. Does this work because we're using `fakeClientSet` so it is not validating that the value of `testNamespace` exists before creating the Deployment with that Namespace?